### ISSUE TITLE
Subscriptions not available for data table?

### ISSUE THREAD

**Issue by brylie** [yogiben/meteor-admin#215]
I am not sure what I am doing wrong, but 'view all' data tables for my custom collections are not populating. Adding Autopublish resolves this, but is not desirable. How can I ensure data is available for collectionsdefined in adminConfig, so that data tables will display rows?

**Comment by mpowaga** [comment]
Are you using `extraFields`? There was a bug but it's solved in the newest version.

**Comment by brylie** [comment]
Yes, I am using `extraFields` on the collections. I will update the code, and close this issue if the issue is resolved. Thanks for your help :-)

**Comment by brylie** [comment]
I should also clarify that the subscriptions also do not seem to be available on the 'add new' forms. I.e. I have several collections with foreign key fields. On the foreign key fields, I am displaying a selection widget pre-populated with options from the related collection.

* [example 1](https://github.com/brylie/juhani-wellbeing/blob/master/both/collections/activities.js#L46)
* [example 2](https://github.com/brylie/juhani-wellbeing/blob/master/both/collections/homes.js#L12)

How can I ensure that related collections, i.e. auxCollections, are available on collection input forms?

**Comment by mpowaga** [comment]
It was a bug. I pushed a fix that will subscribe children documents (defined by `children` property) also on edit page.

**Comment by brylie** [comment]
OK, thanks :-)

**Comment by brylie** [comment]
In what version can I expect this fix to work? I am currently running 1.2.0, and the helper functions do not return any data.

**Comment by brylie** [comment]
Can you please document an example of the children property?

**Comment by mpowaga** [comment]
Check out publish-composite [docs](https://github.com/englue/meteor-publish-composite) if you are not familiar with it yet. You can use children property like this:

```javascript
AdminConfig = {
  collections: {
    Posts: {
      // ...
      children: [
        {
          find: function (post) {
            return Meteor.users.find(post.owner, { limit: 1 });
          }
        }
      ]
    }
  }
};
```

**Comment by brylie** [comment]
In my autoform select options group, I am getting a list of **all** documents in a separate collection, in order to display them in a select menu.  Here is a [quick example](https://github.com/brylie/juhani-wellbeing/blob/master/both/collections/activities.js#L50-L62)

```javascript
autoform: {
      options: function() {
        // Get all documents from another db
        var documents = Documents.find().fetch();

        // Create an options array of documents with label and value pairs
        var documentsOptions = _.map(documents, function(document {
          return {
            label: document.title,
            value: document._id
          };
        });

        return documentsOptions;
      }
    }
```



The publish composite seems to be relevant when the document already contains external document id (such as a foreign key relationship).

**Comment by brylie** [comment]
Essentially, when adding an axillary collection in AdminConfig, it may be necessar to subscribe to all documents in that collection on edit form pages.

**Comment by mpowaga** [comment]
You can publish the whole collection as well, e.g.

```javascript
AdminConfig = {
  collections: {
    Posts: {
      // ...
      children: [
        {
          find: function (post) {
            return Documents.find();
          }
        }
      ]
    }
  }
};
```

**Comment by etcook** [comment]
@mpowaga This does not work for the "new" form. The collection is not exposed, so there are no options when trying to create a new object.

**Comment by brylie** [comment]
Yep, I am still having the issue on the "New" form where related collections are not available. 

I have defined the related collections in the 'auxCollections' and added the 'extraFields' option to include the foreign key field.

**Comment by brylie** [comment]
I created a separate issue (#269) to better describe the auxCollections subscription issue.
